File: testSuite/test-Bolshoi-tree-builder.pl
#!/usr/bin/env perl
use strict;
use warnings;
use Cwd;
use lib $ENV{'GALACTICUS_EXEC_PATH'}."/perl";
use File::Which;

# Take trees from the Bolshoi simulation and process into both Galacticus and IRATE formats.
# Andrew Benson (12-October-2012)
# Contributions to this file made by: Stephanie Dörschner.

# Create output folder.
system("cd ..; mkdir -p testSuite/outputs");

# Create a file in Galacticus format.
system("cd ..; ./Galacticus.exe testSuite/parameters/mergerTreeFileBuildBolshoi.xml");
die("FAILED: failed to make Galacticus-format merger tree file from Bolshoi merger tree")
    unless ( $? == 0 );

# Create a file in IRATE format.
system("cd ..; ./Galacticus.exe testSuite/parameters/mergerTreeFileBuildBolshoiIRATE.xml");
die("FAILED: failed to make IRATE-format merger tree file from Bolshoi merger tree")
    unless ( $? == 0 );

# Validate the IRATE format file.
my $validator = &File::Which::which('iratevalidate');
if ( $validator ) {
    system("cd ..; iratevalidate testSuite/outputs/bolshoiTestTreesIRATE_in.hdf5");
    die("FAILED: IRATE-format file generated from Bolshoi merger tree did not validate")
	unless ( $? == 0 );
} else {
    print "SKIP: iratevalidate is not installed - validation of IRATE-format file will be skipped\n";
}

# Run Galacticus on file in Galacticus format.
system("cd ..; ./Galacticus.exe testSuite/parameters/bolshoiTestTreesGLC.xml");

exit;

File: testSuite/test-Bolshoi-tree-builder.py
#!/usr/bin/env python3
import subprocess
import sys
import shutil

# Take trees from the Bolshoi simulation and process into both Galacticus and IRATE formats.
# Andrew Benson (ported to Python)

# Create output folder.
subprocess.run("cd ..; mkdir -p testSuite/outputs", shell=True)

# Create a file in Galacticus format.
status = subprocess.run("cd ..; ./Galacticus.exe testSuite/parameters/mergerTreeFileBuildBolshoi.xml", shell=True)
if status.returncode != 0:
    print("FAILED: failed to make Galacticus-format merger tree file from Bolshoi merger tree")
    sys.exit(0)

# Create a file in IRATE format.
status = subprocess.run("cd ..; ./Galacticus.exe testSuite/parameters/mergerTreeFileBuildBolshoiIRATE.xml", shell=True)
if status.returncode != 0:
    print("FAILED: failed to make IRATE-format merger tree file from Bolshoi merger tree")
    sys.exit(0)

# Validate the IRATE format file.
validator = shutil.which("iratevalidate")
if validator:
    status = subprocess.run("cd ..; iratevalidate testSuite/outputs/bolshoiTestTreesIRATE_in.hdf5", shell=True)
    if status.returncode != 0:
        print("FAILED: IRATE-format file generated from Bolshoi merger tree did not validate")
        sys.exit(0)
else:
    print("SKIP: iratevalidate is not installed - validation of IRATE-format file will be skipped")

# Run Galacticus on file in Galacticus format.
subprocess.run("cd ..; ./Galacticus.exe testSuite/parameters/bolshoiTestTreesGLC.xml", shell=True)
